syntax = "proto3";

message BrokerCreateTopicRequest {
    string topic_name = 1;
    int32 replication_factor = 2;
}

message BrokerCreateTopicResponse {
    bool success = 1;
}

message BrokerLinkPartitionRequest {
    string partition_name = 1;
    int32 port = 2;
}

message BrokerLinkPartitionResponse {
    bool success = 1;
}

message BrokerProduceRequest {
    string topic_name = 1;
    string message = 2;
}

message BrokerProduceResponse {
    bool success = 1;
}

message BrokerConsumeRequest {
    string topic_name = 1;
}

message BrokerConsumeResponse {
    string message = 1;
}

service BrokerService {
    rpc Produce(BrokerProduceRequest) returns (BrokerProduceResponse);
    rpc Consume(BrokerConsumeRequest) returns (BrokerConsumeResponse);
    rpc CreateTopic(BrokerCreateTopicRequest) returns (BrokerCreateTopicResponse);
    rpc LinkPartition(BrokerLinkPartitionRequest) returns (BrokerLinkPartitionResponse);
}